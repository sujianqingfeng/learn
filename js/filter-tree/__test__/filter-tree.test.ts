import { describe, it, expect } from 'vitest'
import { filterTree, } from '../index'
import type { BadgeTree } from '../index'

describe('filter-tree', () => {
  it('1', () => {

    const original:BadgeTree = {
      'num': 1,
      'link': '',
      'children': [
        {
          'num': 1,
          'link': '/home',
          'children': [
            {
              'num': 1,
              'link': '/home/transcribe',
              'children': [
                {
                  'num': 1,
                  'link': '/home/transcribe/cus-list',
                  'children': [
                    {
                      'num': 1,
                      'link': '/home/transcribe/cus-list/111122222',
                      'children': []
                    }
                  ]
                }
              ]
            }
          ]
        }
      ]
    }

    const whiteRegList = ['', '/home', '/home/transcribe', '/home/transcribe/cus-list', '/home/transcribe/cus-list/.+']

    const result = filterTree(original, whiteRegList)
    expect(result).toMatchSnapshot()
  })

  it('2', () => {
    const original:BadgeTree = {
      'num': 3,
      'link': '',
      'children': [
        {
          'num': 3,
          'link': '/home',
          'children': [
            {
              'num': 3,
              'link': '/home/transcribe',
              'children': [
                {
                  'num': 3,
                  'link': '/home/transcribe/cus-list',
                  'children': [
                    {
                      'num': 1,
                      'link': '/home/transcribe/cus-list/111122222',
                      'children': []
                    },
                    {
                      'num': 1,
                      'link': '/home/transcribe/cus-xxx/111122222',
                      'children': []
                    },
                    {
                      'num': 1,
                      'link': '/home/transcribe/cus-yyy/111122222',
                      'children': []
                    }
                  ]
                }
              ]
            }
          ]
        }
      ]
    }

    const whiteRegList = ['', '/home', '/home/transcribe', '/home/transcribe/cus-list', ['/home/transcribe/cus-xxx/.+', '/home/transcribe/cus-yyy/.+']]

    const result = filterTree(original, whiteRegList)
    expect(result).toMatchSnapshot()
  })
})