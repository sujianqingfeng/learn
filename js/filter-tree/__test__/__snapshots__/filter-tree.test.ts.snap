// Vitest Snapshot v1

exports[`filter-tree > 1 1`] = `
{
  "children": [
    {
      "children": [
        {
          "children": [
            {
              "children": [
                {
                  "children": [],
                  "link": "/home/transcribe/cus-list/111122222",
                  "num": 1,
                },
              ],
              "link": "/home/transcribe/cus-list",
              "num": 1,
            },
          ],
          "link": "/home/transcribe",
          "num": 1,
        },
      ],
      "link": "/home",
      "num": 1,
    },
  ],
  "link": "",
  "num": 1,
}
`;

exports[`filter-tree > 2 1`] = `
{
  "children": [
    {
      "children": [
        {
          "children": [
            {
              "children": [
                {
                  "children": [],
                  "link": "/home/transcribe/cus-xxx/111122222",
                  "num": 1,
                },
                {
                  "children": [],
                  "link": "/home/transcribe/cus-yyy/111122222",
                  "num": 1,
                },
              ],
              "link": "/home/transcribe/cus-list",
              "num": 2,
            },
          ],
          "link": "/home/transcribe",
          "num": 3,
        },
      ],
      "link": "/home",
      "num": 3,
    },
  ],
  "link": "",
  "num": 3,
}
`;
